### PR TITLE
[set-webpack-public-path-plugin] Add functionality to use the webpack-produced asset name to find the script path.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: webpack/localization-plugin/src/LocalizationPlugin.ts [hunk 1]
```diff
@@ -157,43 +159,51 @@ export class LocalizationPlugin implements Webpack.Plugin {
...
-                const localizedChunkAssets: { [locale: string]: string } = {};
-                for (const [locale, newAsset] of resultingAssets) {
-                  compilation.assets[newAsset.filename] = newAsset.asset;
-                  localizedChunkAssets[locale] = newAsset.filename;
-                }
+            for (const chunk of chunkGroup.chunks) {
+              // Clone the chunk files array because we're going to modify it
+              const chunkFiles: string[] = [...chunk.files];
+              const chunkFilesSet: Set<string> = new Set(chunkFiles);
+              for (const chunkFileName of chunkFiles) {
```
Comment (apostolisms) [inline webpack/localization-plugin/src/LocalizationPlugin.ts]: can this be optimized?

File: webpack/set-webpack-public-path-plugin/src/SetPublicPathPlugin.ts [hunk 2]
```diff
@@ -127,48 +147,79 @@ export class SetPublicPathPlugin implements Webpack.Plugin {
...
+            });
+          } else {
+            return source;
+          }
+        });
+      });
+
+      compiler.hooks.emit.tap(PLUGIN_NAME, (compilation: Webpack.compilation.Compilation) => {
+        for (const chunkGroup of compilation.chunkGroups) {
+          for (const chunk of chunkGroup.chunks) {
```
Comment (apostolisms) [inline webpack/set-webpack-public-path-plugin/src/SetPublicPathPlugin.ts]: can this be optimized?

File: webpack/localization-plugin/src/LocalizationPlugin.ts (hunk 1 above)
Comment (iclanton) [reply webpack/localization-plugin/src/LocalizationPlugin.ts]: Not really. It'll still end up iterating through the same number of elements, regardless of how we do the iteration.

File: webpack/set-webpack-public-path-plugin/src/SetPublicPathPlugin.ts (hunk 2 above)
Comment (iclanton) [reply webpack/set-webpack-public-path-plugin/src/SetPublicPathPlugin.ts]: Same story - we still need to loop through each chunk-produced asset, so regardless of how we loop through them, it'll be the same number of elements.

**Pull request merged**